prow/gerrit/adapter/adapter.go

```diff
@@ -91,38 +91,52 @@ func NewController(ctx context.Context, prowJobClient prowv1.ProwJobInterface, o
 	if err != nil {
 		logrus.WithError(err).Fatal("Error creating gerrit client.")
 	}
-	gerritClient.Authenticate(cookiefilePath, tokenPathOverride)
-
 	c := &Controller{
 		prowJobClient:      prowJobClient,
 		config:             cfg,
 		gc:                 gerritClient,
 		tracker:            lastSyncTracker,
 		projectsOptOutHelp: projectsOptOutHelpMap,
 	}
+
+	// applyGlobalConfig must happen before authentication, otherwise initially
+	// the gerritClient doesn't contain any repos, and nothing is authenticated.
+	// The async goroutine for authentication only does something in real when httpcookiefile
+	// is updated.
+	c.applyGlobalConfig(cfg, gerritClient, lastSyncTracker)
+	gerritClient.Authenticate(cookiefilePath, tokenPathOverride)
```

Comment:
Don't we need to call this after every call to `gerritClient.UpdateClients()` then?

Reply:
that's a good point, that would result in multiple goroutines doing async authentication though, will think about how to do this better

Reply:
actually I read it wrong, `Authenticate` is smart enough to only spin off goroutine once. Updated